test/lib/pmd/PmdCatalogWrapper.test.ts

```diff
@@ -90,4 +97,27 @@ describe('PmdCatalogWrapper', () => {
 			});
 		});
 	});
+
+	describe('getRulePathEntries()', () => {
+		before(() => {
+			Sinon.createSandbox();
+			// Stub out the array so that it returns 'javascript'
+			Sinon.stub(PmdLanguageManager, "getSupportedLanguages").resolves([LANGUAGES.APEX, LANGUAGES.JAVASCRIPT]);
+		});
+
+		after(() => {
+			Sinon.restore();
+		});
+
+		it('Throws exception if javascript is found in supportedLanguages array', async () => {
+			const target = await TestablePmdCatalogWrapper.create({});
+
+			try {
```

Comment:
There is a `chai-as-promised` plugin which would clean up the syntax here a little, but I didn't want to add a new dependency late in the game.

Reply:
I've been looking into it as well and the `.beRejected` clause seems particularly useful.
Agree, let's add it for Scanner3.0.